test/bin/scenario.sh

```diff
@@ -1192,9 +1227,64 @@ action_run() {
     record_junit "run" "scenario_run_tests" "OK"
 }
 
+action_run_on_vm() {
```

Comment:
I wonder if it's possible to avoid this duplication of the run logic for this specific case. 
If we only need to override the VM IP, can we provide an optional argument (e.g. `--ip_override 1.2.3.4`) and set it un the current `run_tests` function if specified.